### PR TITLE
Clear out ElasticResponse large hits collection after done materializing.

### PR DESCRIPTION
**Pull request by marcind**
A large enough hits list will end up getting allocated on the large object heap, which will pin all the result documents in memory longer than necessary.

This is one of the changes to address issues mentioned in https://github.com/CenturyLinkCloud/ElasticLINQ/issues/89

### CODE REVIEW

File: Source/ElasticLINQ/ElasticQueryProvider.cs [hunk 1]
```diff
@@ -123,7 +123,22 @@ public object Execute(Expression expression)
                         throw new InvalidOperationException("No HTTP response received.");
                 }
 
-                return translation.Materializer.Materialize(response);
+                var result = translation.Materializer.Materialize(response);
+
+                if (response.hits != null)
+                {
+                    var hits = response.hits.hits;
+                    if (hits != null && hits.Capacity > 4096)
```
Comment (damieng) [inline Source/ElasticLINQ/ElasticQueryProvider.cs:131]: Is there any harm in just always clearing the hits after materialization?

Comment (marcind) [inline Source/ElasticLINQ/ElasticQueryProvider.cs:131]: Probably not. I guess zeroing out an array should be fairly well optimized in .net.

**Comment by damieng** [comment]
The following change would probably be better for memory... but worse for perf (in ListHitsElasticMaterializer)

``` csharp
        internal static IEnumerable<T> Many<T>(List<Hit> hits, Func<Hit, object> projector)
        {
            while (hits.Count > 0)
            {
                var result = (T)projector(hits[0]);
                hits.RemoveAt(0);
                yield return result;
            }
        }
```

Thoughts?

**Comment by marcind** [comment]
Could probably write a quick app to compare, but my intuition tells me it'll be pretty bad since `RemoveAt` shifts all the remaining items to the left, so there's O(n^2) memory being moved around.

**Pull request merged**